include/pybind11/numpy.h

```diff
@@ -507,11 +507,16 @@ class dtype : public object {
         return detail::array_descriptor_proxy(m_ptr)->names != nullptr;
     }
 
-    /// Single-character type code.
+    /// Single-character for dtype's kind (ex: float and double are 'f' or int and long int are 'i')
     char kind() const {
         return detail::array_descriptor_proxy(m_ptr)->kind;
     }
 
+    /// Single-character for dtype's type (ex: float is 'f' and double 'd')
+    char char_() const {
+        return detail::array_descriptor_proxy(m_ptr)->type;
```

Comment:
Is it worth adding a comment that we're following the Python API of `dtype` (i.e., `dtype.char`), rather than NumPy's internal C API (`PyArray_Descr::type`), for the name?

See also https://numpy.org/devdocs/reference/c-api/types-and-structures.html#c.PyArray_Descr for the root cause of the `type` vs. `char` confusion.

Reply:
okay !